### ISSUE TITLE
'TypeError: icon is undefined' fatal error when started with "Apps Icon" unchecked

### ISSUE THREAD

**Issue by joshqou** [icedman/dash2dock-lite#99]
Adding `icon &&` at dock.js#LN462 fixes the startup crash. Not sure where `_showAppsIcon` is defined so can't investigate further

```
Extension dash2dock-lite@icedman.github.com: TypeError: icon is undefined

Stack trace:
  _findIcons@file:///home/jqou/.local/share/gnome-shell/extensions/dash2dock-lite@icedman.github.com/dock.js:462:13
  layout@file:///home/jqou/.local/share/gnome-shell/extensions/dash2dock-lite@icedman.github.com/dock.js:682:26
  dock@file:///home/jqou/.local/share/gnome-shell/extensions/dash2dock-lite@icedman.github.com/dock.js:137:12
  createDock@file:///home/jqou/.local/share/gnome-shell/extensions/dash2dock-lite@icedman.github.com/extension.js:51:7
  createTheDocks@file:///home/jqou/.local/share/gnome-shell/extensions/dash2dock-lite@icedman.github.com/extension.js:71:14
  startUp@file:///home/jqou/.local/share/gnome-shell/extensions/dash2dock-lite@icedman.github.com/extension.js:222:10
  enable@file:///home/jqou/.local/share/gnome-shell/extensions/dash2dock-lite@icedman.github.com/extension.js:163:10
  _callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:253:38
  _onEnabledExtensionsChanged@resource:///org/gnome/shell/ui/extensionSystem.js:606:24
  _loadExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:705:18
  createCheckedMethod/<@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:730:46
  enableExtension@resource:///org/gnome/shell/ui/extensionSystem.js:277:29
  EnableExtension@resource:///org/gnome/shell/ui/shellDBus.js:453:38
  _handleMethodCall@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:329:38
  _wrapJSObject/<@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:408:34
  @resource:///org/gnome/shell/ui/init.js:21:20
  
```

**Comment by icedman** [comment]
Pushed the fix at the development branch. Thanks for the report. Including the stack trace really does help.

**Comment by joshqou** [comment]
Fixed by ce6d389
